src/client.rs

```diff
@@ -85,19 +85,22 @@ impl SerializeMessage for str {
 //TODO add more DeserializeMessage impls
 
 #[derive(Clone)]
-pub struct Pulsar {
-    manager: Arc<ConnectionManager>,
+pub struct Pulsar<P> {
```

Comment:
very minor nit: I generally like to name the generic closer to the true generic thing, so in this case its an executor so `E` makes much more sense to me.

Reply:
This should be resolved now